compute/instances/troubleshooting/fix-dns-routed-ipv6-only-debian-bullseye.mdx

```diff
@@ -0,0 +1,147 @@
+---
+meta:
+  title: Fixing DNS resolution with a routed IPv6-only setup on Debian Bullseye
+  description: This page helps you to fix DNS resolution with a routed IPv6-only setup on Debian Bullseye
+content:
+  h1: Fixing DNS resolution with a routed IPv6-only setup on Debian Bullseye
+  paragraph: This page helps you to fix DNS resolution with a routed IPv6-only setup on Debian Bullseye
+tags: dns ipv6 Instance debian bullseye
+dates:
+  validation: 2024-01-23
+  posted: 2024-01-23
+categories:
+  - compute
+---
+
+This guide outlines the steps to enable DNS resolution on a Scaleway Instance that uses the Debian Bullseye image, configured with one or more routed IPv6 addresses and without any IPv4 addresses.
+
+<Message type="requirement">
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [an Instance](/compute/Instances/how-to/create-an-Instance) running Debian Bullseye (11) with an IPv6-only setup
+</Message>
+
+<Message type="important">
+    This guide explains how to switch from the traditional `ifupdown` toolset to the more modern `netplan` solution, which is used in recent Debian releases. The process is straightforward if your Instance is using the default network configuration provided by Scaleway metadata. 
+    However, if your Instance has a customized network setup, ensure you are comfortable with both `ifupdown` and `netplan` before proceeding. You can safely apply this procedure to a newly created Instance.
+</Message>
+
+## Technical explanation
+
+When a Scaleway Instance uses routed IP addresses, the IPv6 stack is automatically configured using [`SLAAC`](https://datatracker.ietf.org/doc/html/rfc4862). With this method, the Instance is periodically advertised with various network configuration details, including the DNS server addresses it should use. The Instance is then free to consume these advertisements or not. By default, the operating system images provided by Scaleway are configured to leverage these advertisements to configure the IPv6 networking and the related DNS servers. The Debian Bullseye image is no exception.
+
+When configuring the network at boot time, the `cloud-init` software detects the appropriate network configuration method used by the system at hand and write and/or apply the necessary configuration files/parameters. On Debian Bullseye, and because of [`cloud-init`'s built-in order of detection](https://cloudinit.readthedocs.io/en/latest/reference/network-config.html#network-output-policy), the primarily detected method is [ENI](https://cloudinit.readthedocs.io/en/latest/reference/network-config-format-eni.html), which configures the network through Debian's well known `/etc/network/interfaces*` set of files, along with the `ifupdown` toolset.
```

Comment:
```suggestion
When configuring the network at boot time, the `cloud-init` software detects the appropriate network configuration method used by the system at hand and writes and/or applies the necessary configuration files/parameters. On Debian Bullseye, and because of [`cloud-init`'s built-in order of detection](https://cloudinit.readthedocs.io/en/latest/reference/network-config.html#network-output-policy), the primary detected method is [ENI](https://cloudinit.readthedocs.io/en/latest/reference/network-config-format-eni.html), which configures the network through Debian's well known `/etc/network/interfaces*` set of files, along with the `ifupdown` toolset.
```